guvnor-structure/guvnor-structure-client/src/main/java/org/guvnor/structure/client/editors/repository/clone/CloneRepositoryPresenter.java

```diff
@@ -0,0 +1,211 @@
+                                    } catch ( Throwable ex ) {
+                                        view.errorCloneRepositoryFail(ex);
+                                    }
+                                    unlockScreen();
+                                    return true;
+                                }
+                            }
+                    ).createRepository( availableOrganizationalUnits.get( organizationalUnit ), scheme, alias, env );
+
+                }
+            } ).normalizeRepositoryName( view.getName() );
+        }
+    }
+
+    private void populateOrganizationalUnits() {
+        //populate Organizational Units list box
+        organizationalUnitService.call(
+            new RemoteCallback<Collection<OrganizationalUnit>>() {
+                @Override
+                public void callback(Collection<OrganizationalUnit> organizationalUnits) {
+                    view.addOrganizationalUnitSelectEntry();
+                    if (organizationalUnits != null && !organizationalUnits.isEmpty()) {
+                        for (OrganizationalUnit organizationalUnit : organizationalUnits) {
+                            view.addOrganizationalUnit(organizationalUnit.getName(),
+                                    organizationalUnit.getName());
+                            availableOrganizationalUnits.put(organizationalUnit.getName(),
+                                    organizationalUnit);
+                        }
+                    }
+                }
+            },
+            new ErrorCallback<Message>() {
+                @Override
+                public boolean error(final Message message,
+                                     final Throwable throwable) {
+                    view.errorLoadOrganizationalUnitsFail(throwable);
+                    return false;
+                }
+            }
+        ).getOrganizationalUnits();
+    }
+
+    public void showForm() {
+        view.showForm();
+    }
+
+    private void lockScreen() {
+        view.showBusyPopupMessage();
+        view.setPopupCloseVisible( false );
+        view.setCloneEnabled( false );
+        view.setCancelEnabled( false );
+        view.setPasswordEnabled( false );
+        view.setUsernameEnabled( false );
+        view.setGitUrlEnabled( false );
+        view.setOrganizationalUnitEnabled( false );
+        view.setNameEnabled(false);
+    }
+
+    private void unlockScreen() {
+        view.closeBusyPopup();
+        view.setPopupCloseVisible(true);
+        view.setCloneEnabled(true);
+        view.setCancelEnabled(true);
+        view.setPasswordEnabled(true);
+        view.setUsernameEnabled(true);
+        view.setGitUrlEnabled(true);
+        view.setOrganizationalUnitEnabled( true );
+        view.setNameEnabled(true);
+    }
+
+    protected boolean isOuMandatory() {
+        try {
+            final IOCBeanDef<RepositoryPreferences> beanDef = IOC.getBeanManager().lookupBean(RepositoryPreferences.class);
```

Comment:
You can ```@Inject``` an instance of ```RepositoryPreferences``` rather than use ```IOC``` here. The old code used ```IOC``` as the old code was not CDI-enabled and hence we needed to instantiate the ```RepositoryPreferences``` object with the IOC Bean Manager manually.